File: FSM/Entity/QuoteOrderProducts/QuoteOrderProduct.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.QuoteOrderProducts.QuoteOrderProduct
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using System.Collections;
using System.Runtime.CompilerServices;

namespace FSM.Entity.QuoteOrderProducts
{
  public class QuoteOrderProduct
  {
    private DataTypeValidator _dataTypeValidator;
    private bool _exists;
    private bool _deleted;
    private int _QuoteOrderProductID;
    private int _QuoteOrderID;
    private int _ProductID;
    private int _Quantity;
    private double _Price;

    public QuoteOrderProduct()
    {
      this._exists = false;
      this._deleted = false;
      this._QuoteOrderProductID = 0;
      this._QuoteOrderID = 0;
      this._ProductID = 0;
      this._Quantity = 0;
      this._Price = 0.0;
      this._dataTypeValidator = new DataTypeValidator();
    }

    public bool IgnoreExceptionsOnSetMethods
    {
      get
      {
        return this._dataTypeValidator.IgnoreExceptionsOnSetMethods;
      }
      set
      {
        this._dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
      }
    }

    public Hashtable Errors
    {
      get
      {
        return this._dataTypeValidator.Errors;
      }
    }

    public DataTypeValidator DTValidator
    {
      get
      {
        return this._dataTypeValidator;
      }
    }

    public bool Exists
    {
      get
      {
        return this._exists;
      }
      set
      {
        this._exists = value;
      }
    }

    public bool Deleted
    {
      get
      {
        return this._deleted;
      }
    }

    public bool SetDeleted
    {
      set
      {
        this._deleted = value;
      }
    }

    public int QuoteOrderProductID
    {
      get
      {
        return this._QuoteOrderProductID;
      }
    }

    public object SetQuoteOrderProductID
    {
      set
      {
        this._dataTypeValidator.SetValue((object) this, "_QuoteOrderProductID", RuntimeHelpers.GetObjectValue(value));
      }
    }

    public int QuoteOrderID
    {
      get
      {
        return this._QuoteOrderID;
      }
    }

    public object SetQuoteOrderID
    {
      set
      {
        this._dataTypeValidator.SetValue((object) this, "_QuoteOrderID", RuntimeHelpers.GetObjectValue(value));
      }
    }

    public int ProductID
    {
      get
      {
        return this._ProductID;
      }
    }

    public object SetProductID
    {
      set
      {
        this._dataTypeValidator.SetValue((object) this, "_ProductID", RuntimeHelpers.GetObjectValue(value));
      }
    }

    public int Quantity
    {
      get
      {
        return this._Quantity;
      }
    }

    public object SetQuantity
    {
      set
      {
        this._dataTypeValidator.SetValue((object) this, "_Quantity", RuntimeHelpers.GetObjectValue(value));
      }
    }

    public double Price
    {
      get
      {
        return this._Price;
      }
    }

    public object SetPrice
    {
      set
      {
        this._dataTypeValidator.SetValue((object) this, "_Price", RuntimeHelpers.GetObjectValue(value));
      }
    }
  }
}

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteOrder/QuoteOrderProduct/QuoteOrderProduct.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace QuoteOrderProducts

Public Class QuoteOrderProduct

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "QuoteOrderProduct Properties"

      
Private _QuoteOrderProductID As Integer = 0
Public Readonly Property QuoteOrderProductID() As Integer
	Get 
		Return _QuoteOrderProductID
	End Get	
End Property	
Public Writeonly Property SetQuoteOrderProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteOrderProductID", Value)
	End Set
End Property


Private _QuoteOrderID As Integer = 0
Public Readonly Property QuoteOrderID() As Integer
	Get 
		Return _QuoteOrderID
	End Get	
End Property	
Public Writeonly Property SetQuoteOrderID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_QuoteOrderID", Value)
	End Set
End Property


Private _ProductID As Integer = 0
Public Readonly Property ProductID() As Integer
	Get 
		Return _ProductID
	End Get	
End Property	
Public Writeonly Property SetProductID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_ProductID", Value)
	End Set
End Property

            Private _Quantity As Integer = 0
            Public ReadOnly Property Quantity() As Integer
                Get
                    Return _Quantity
                End Get
            End Property
            Public WriteOnly Property SetQuantity() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Quantity", Value)
                End Set
            End Property

            Private _Price As Double = 0
            Public ReadOnly Property Price() As Double
                Get
                    Return _Price
                End Get
            End Property
            Public WriteOnly Property SetPrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Price", Value)
                End Set
            End Property

#End Region

End Class

End Namespace

End Namespace

